protocol/src/BridgeDepositWithdraw.sol

```diff
@@ -0,0 +1,294 @@
+            // Store selector in first 4 bytes
+            mstore(ptr, selector)
+            // Store address (32 bytes, right-padded) starting at byte 4
+            mstore(add(ptr, 0x04), addr)
+            result := keccak256(ptr, 0x24) // 36 bytes = 0x24
+        }
+    }
+
+    /// @dev Hashes deposit input: selector + token + amount + to (100 bytes) using inline assembly
+    /// Equivalent to keccak256(abi.encodePacked(bytes4(keccak256("deposit(address,uint256,address)")), abi.encode(token, amount, to)))
+    function _hashDepositInput(address token, uint256 amount, address to) internal pure returns (bytes32 result) {
+        bytes4 selector = bytes4(keccak256("deposit(address,uint256,address)"));
+        assembly {
+            let ptr := mload(0x40)
+            // Store selector in first 4 bytes
+            mstore(ptr, selector)
+            // Store token (32 bytes) starting at byte 4
+            mstore(add(ptr, 0x04), token)
+            // Store amount (32 bytes) starting at byte 36
+            mstore(add(ptr, 0x24), amount)
+            // Store to (32 bytes) starting at byte 68
+            mstore(add(ptr, 0x44), to)
+            result := keccak256(ptr, 0x64) // 100 bytes = 0x64
+        }
+    }
+
+    /**
+     * @inheritdoc IBridgeDepositWithdraw
+     */
+    function claim(
+        address token,
+        uint256 amount,
+        address to,
+        uint64 committeeEpoch,
+        bytes calldata aggregatedSignatures,
+        MerkleTree.MultiProof calldata proof
+    ) public override whenNotPaused {
+        // Check if token is configured - mirrorTokens stores the local token for a given source token
+        // A token is configured if it has tokenData with minAmount > 0
+        address mirrorToken = mirrorTokens[token];
+        // For token-to-native: mirrorToken will be MOCK_ADDRESS_FOR_NATIVE_DEPOSIT
+        // For token-to-token: mirrorToken will be an ERC20 address
+        if (mirrorToken == address(0)) revert MirrorTokenNotFound();
+        checkValidClaim(mirrorToken, amount);
+
+        // Build leaves for validating merkle proof (no 'from' verification - anyone can claim)
+        bytes32[] memory leaves = new bytes32[](2);
+        leaves[0] = MerkleTree.hashLeaf("to", BRIDGE_CONTRACT_HASH);
+        leaves[1] = MerkleTree.hashLeaf("input", _hashDepositInput(token, amount, to));
+
+        sortLeaves(leaves);
+
+        // Compute the merkle root from leaves and proof
+        bytes32 txHash = MerkleTree.processMulti(leaves, proof);
+
+        // Check if already processed
+        if (processedRequests[txHash]) revert RequestAlreadyProcessed();
+
+        // Reconstruct AttestedTx and verify signatures
+        bytes32 attestedHash = AttestedTx.digest(txHash, committeeEpoch);
+        bytes32 signedHash = _hashBytes32(attestedHash);
+        address[] memory validators = ECDSA.recoverSigners(signedHash, aggregatedSignatures);
+
+        uint256 weight = podConfig.registry.computeWeight(validators);
```

Comment:
it might be much cheaper for the bridge to locally have the set of signers and register on the registry on contract to receive updates whenever the committee changes (push vs pull). 

then we probably also never even create a list of validators that signed, just go over the signatures and update a counter and bitmask (for uniqueness)